auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailActivity.java

```diff
@@ -143,13 +289,13 @@ public void onComplete(@NonNull Task<Void> task) {
                                         // the credential to SmartLock (if enabled).
                                         mActivityHelper.saveCredentialsOrFinish(
                                                 mSaveSmartLock,
-                                                firebaseUser,
+                                                user,
                                                 password);
                                     }
                                 });
                     }
                 })
-                .addOnFailureListener(new OnFailureListener() {
+                .addOnFailureListener(this, new OnFailureListener() {
```

Comment:
Here was my logic:

If the request succeeds, we to continue whether or not the original activity is still there because `mSaveSmartLock` is already initialized and not tied to the activity lifecycles. That way, the activity will finish and the sign in will proceed.

If it fails, the only thing we are doing in the failure listener is updating views. I'm not sure how that works, but I would think something bad will happen if we try to update a view from a previous instance of an activity. Am I wrong?